.github/mergify.yml

```diff
@@ -0,0 +1,42 @@
+pull_request_rules:
+- name: auto-merge
+  description: automatic merge for main with > 1 approved reviews, all requested reviews have given feedback, not held, and CI is successful
+  conditions:
+    - "#approved-reviews-by>=1"
+    - "#review-requested=0"
+    - "#changes-requested-reviews-by=0"
+    - base=main
+    - label!=hold
+    - check-success=DCO
+
+    # If files are changed in .github/, the actionlint check must pass
+    - or:
+      - and:
+        # regex should match the one in .github/workflows/actionlint.yml
+        - files~=.github/.*$
+        - check-success=actionlint
+      - -files~=.github/.*$
+
+    # lint and test must pass if files change that would trigger this job
+    - or:
+      - and:
+        - check-success=lint
+        - check-success=test
+        - or:
+          # see .github/workflows/lint.yml and test.yml
+          - files~=.*\.py$
+          - files~=pyproject.toml$
+          - files~=requirements.*\.txt$
+          - files~=tox.ini$
+          - files~=scripts/.*\.sh$
+      - and:
+        - -files~=.github/.*$
+        - -files~=.*\.py$
+        - -files~=pyproject.toml$
+        - -files~=requirements.*\.txt$
+        - -files~=tox.ini$
+        - -files~=scripts/.*\.sh$
+
+  actions:
+    merge:
+      method: merge
```

Comment:
@leseb If I opened up a PR to say the CLI repo from a branch on my fork would configuring `delete_head_branch: {}` delete the branch on my fork of CLI automatically? I'd rather bots stay out of my fork's branches.

Reply:
it would delete it only after merging it, which is typically what most people want.

I'm also fine *not* turning that on here. I don't really care what branches are left around in people's forks.

Reply:
After more consideration, I'm in favor of adding that feature here.

Reply:
Well you got in my head, and now I don't know if I'm in favor of it!

Maybe it's worth doing as a follow-up PR in any case.

Reply:
Still +1 on this :). I believe this is what we want most of the time, it's cosmetic but helps keep a clean repository.

Reply:
Honestly I usually enable this for repos as it's also something you can do in GitHub settings - I say do it but if anyone doesn't want it agreed on the point that branches in people's forks aren't a huge concern for me

Reply:
I added this.